lib/puppet/type/sensu_user.rb

```diff
@@ -44,7 +44,7 @@
   newparam(:name, :namevar => true) do
     desc "The name of the user."
     validate do |value|
-      unless value =~ /^[\w\.\-]+$/
+      unless value =~ %r{^[\w.\-]+$}
```

Comment:
I thought users can't have `.` or `:` in the username

Reply:
should this be a type?

Reply:
They can not, this regex matches regex from Sensu Go which allows period but not colon.

This is a one-off regex not shared by other types so kept separate.